pkg/filter/proxy/pool.go

```diff
@@ -216,7 +242,9 @@ func (p *pool) doRequest(ctx context.HTTPContext, req *request, client *http.Cli
 	}
 
 	span := ctx.Span().NewChildWithStart(spanName, req.startTime())
-	span.Tracer().Inject(span.Context(), opentracing.HTTPHeaders, opentracing.HTTPHeadersCarrier(req.std.Header))
+	if !span.IsNoopSpan() {
+		span.Tracer().Inject(span.Context(), opentracing.HTTPHeaders, opentracing.HTTPHeadersCarrier(req.std.Header))
```

Comment:
Even this line was executed, it would be a dummy operation if the span is noopspan, right?

Reply:
Yes, it would be no-ops operations from here https://github.com/opentracing/opentracing-go/blob/master/noop.go

Reply:
So can this if-clause be omitted?

Reply:
Good point, I omitted it!